test/common/filesystem/directory_test.cc

```diff
@@ -43,11 +54,26 @@ class DirectoryTest : public testing::Test {
   void addFiles(std::list<std::string> files) {
     for (const std::string& file_name : files) {
       const std::string full_path = dir_path_ + "/" + file_name;
-      { const std::ofstream file(full_path); }
+      {
+        const std::ofstream file(full_path);
+        EXPECT_TRUE(file) << "failed to open test file";
+      }
       files_to_remove_.push(full_path);
     }
   }
 
+  void addFileWithContents(absl::string_view file_name, absl::string_view contents) {
+    const std::string full_path = absl::StrCat(dir_path_, "/", file_name);
+    {
+      std::ofstream file(full_path);
+      EXPECT_TRUE(file) << "failed to open test file";
+      file << contents;
+      file.close();
+      EXPECT_TRUE(file) << "failed to write to test file";
```

Comment:
sorry ignorance of ostream but if you fail to write will file become null/false?